File: src/main/java/com/linebot/model/common/RequestParameter.java
package com.linebot.model.common;

import org.springframework.util.MultiValueMap;

import java.util.Objects;

public interface RequestParameter {
    public MultiValueMap<String, String> convert();
    default public void add(MultiValueMap<String, String> param, String keyName, Object value) {
        if (value != null) {
            param.add(keyName, Objects.toString(value));
        }
    }
    default public void required(String keyName, Object value) {
        if (value == null) {
            throw new IllegalArgumentException(keyName + " is required.");
        }
    }
}

File: src/main/kotlin/com/linebot/model/common/RequestParameter.kt
package com.linebot.model.common

import java.util.Objects
import org.springframework.util.MultiValueMap

interface RequestParameter {
    fun convert(): MultiValueMap<String, String>
    fun add(param: MultiValueMap<String, String>, keyName: String, value: Any?) {
        value?.let {
            param.add(keyName, Objects.toString(it))
        }
    }

    fun required(keyName: String, value: Any?) {
        if (value == null) {
            throw IllegalArgumentException("$keyName is required.")
        }
    }
}
